components/data-table/private/head.jsx

```diff
@@ -13,295 +13,304 @@ import HeaderCell from './header-cell';
 import InteractiveElement from '../interactive-element';
 import CellContext from '../private/cell-context';
 import TableContext from '../private/table-context';
-import contextHelper from './context-helper';
+import useContextHelper from './context-helper';
 
 // ## Constants
 import { DATA_TABLE_HEAD } from '../../../utilities/constants';
 
 const InteractiveCheckbox = InteractiveElement(Checkbox);
 
-/**
- * Used internally, provides header row rendering to the DataTable.
- */
-class DataTableHead extends React.Component {
-	// ### Display Name
-	// Always use the canonical component name as the React display name.
-	static displayName = DATA_TABLE_HEAD;
+// ### Prop Types
+const propTypes = {
+	assistiveText: PropTypes.shape({
+		actionsHeader: PropTypes.string,
+		columnSort: PropTypes.string,
+		columnSortedAscending: PropTypes.string,
+		columnSortedDescending: PropTypes.string,
+		selectAllRows: PropTypes.string,
+		selectRow: PropTypes.string,
+	}),
+	allSelected: PropTypes.bool,
+	headerRefs: PropTypes.func,
+	isHidden: PropTypes.bool,
+	indeterminateSelected: PropTypes.bool,
+	canSelectRows: PropTypes.oneOfType([
+		PropTypes.bool,
+		PropTypes.oneOf(['checkbox', 'radio']),
+	]),
+	columns: PropTypes.arrayOf(
+		PropTypes.shape({
+			Cell: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
+			props: PropTypes.object,
+		})
+	),
+	fixedHeader: PropTypes.bool,
+	id: PropTypes.string,
+	onToggleAll: PropTypes.func,
+	onSort: PropTypes.func,
+	showRowActions: PropTypes.bool,
+};
+
+const ActionsHeader = (props) => {
```

Comment:
Extracted the `ActionsHeader` and `SelectHeader` subcomponents since otherwise the hook calls were under a condition which is not allowed. (and also just seemed a good thing to do in general)